### PR TITLE
Feat/pull precheck

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves https://github.com/ubiquity-os/plugins-wishlist/issues/45
Requires #9 
Requires #14

TODO

### CODE REVIEW

**Comment by Keyrxng** [comment]
current QA: https://github.com/ubq-testing/ask-plugin/pull/5#issuecomment-2434179314

File: src/handlers/pull-precheck.ts
Comment (Keyrxng) [inline src/handlers/pull-precheck.ts:55]: I've inadvertently implemented a standard that we enforce every single PR must have a linked task, a happy byproduct I'd imagine, should I keep this?

- merging to main, we bypass
- any PR opened by a non-human contributor, we bypass

**Comment by Keyrxng** [comment]
Bulk of the logic for this is written, it's refining the prompts that's left now. 

Previously you have had prompt structures in mind especially for formatting the context window, do you have anything in mind for this feature?

File: src/handlers/find-ground-truths.ts
Comment (0x4007) [inline src/handlers/find-ground-truths.ts:74]: When dealing with code, we should use Claude actually.

File: src/handlers/llm-query-output.ts
Comment (0x4007) [inline src/handlers/llm-query-output.ts:15]: We have some means to add structured metadata via a method from the SDK

File: src/handlers/pull-precheck.ts
Comment (0x4007) [reply src/handlers/pull-precheck.ts:55]: There have been exceptions that come to mind. Usually with collaborators but occasionally with proactive contributors. This should not be enforced.

File: src/handlers/pull-precheck.ts
Comment (0x4007) [inline src/handlers/pull-precheck.ts:82]:
```suggestion
    throw logger.error("Task specification not found, please link one before merging.");
```
Unnecessary shorthand

File: src/handlers/pull-precheck.ts
Comment (0x4007) [inline src/handlers/pull-precheck.ts:87]: This should be in a test file, or this pull should be a draft

File: src/handlers/pull-precheck.ts
Comment (0x4007) [inline src/handlers/pull-precheck.ts:93]:
```suggestion
  const question = "Does the code diff implement all of the required changes per the specification?";
```
Not sure the context of how this is used but i would also go on to provide instructions for how it should offer feedback.

File: src/plugin.ts
Comment (0x4007) [inline src/plugin.ts:4]: This package name doesnt seem right as we officially deprecated ubiquibot some weeks ago

File: src/types/pull-requests.ts
Comment (0x4007) [inline src/types/pull-requests.ts:1]: Is it really necessary that it needs its own file

**Review by 0x4007 — CHANGES_REQUESTED** [review]

**Comment by 0x4007** [comment]
> Bulk of the logic for this is written, it's refining the prompts that's left now. 
> 
> 
> 
> Previously you have had prompt structures in mind especially for formatting the context window, do you have anything in mind for this feature?

I'm not sure what you're referring to.

File: src/handlers/find-ground-truths.ts
Comment (Keyrxng) [reply src/handlers/find-ground-truths.ts:74]: This is taking ground truths from the task spec not the codebase but I'll use claude for code parsing sure.

File: src/handlers/pull-precheck.ts
Comment (Keyrxng) [reply src/handlers/pull-precheck.ts:93]: I haven't started refining the actual review prompt or anything close to it yet so this is not implemented yet

File: src/types/pull-requests.ts
Comment (Keyrxng) [reply src/types/pull-requests.ts:1]: If by the time I'm done it's still solo I'll group it but I expect to fill more types here

**Comment by Keyrxng** [comment]
> But its important to use clear headers to divide the sections.

https://github.com/ubiquity-os/plugins-wishlist/issues/29#issue-2402769945 - You were very specific and intentional when I implemented `/gpt` which is why I asked but no worries ty

File: src/plugin.ts
Comment (gentlementlegen) [reply src/plugin.ts:4]: @0x4007 We should delete these to avoid confusion, they are still accessible within NPM.

Comment (0x4007) [reply src/plugin.ts:4]: I cant now hope to in some days.

**Comment by Keyrxng** [comment]
Sitting on this until my other PRs in this repo are merged at least while I push forward my other tasks that don't have dependencies.

This PR uses both features introduced in #9 and #14.

**Comment by Keyrxng** [comment]
Didn't realise I hadn't pushed code from 10 days ago and left a comment on `tools` "probs not a good idea to have these for the chatbot" lmao but I took from my early iteration and haven't upgraded it yet/

So anyway tools need an interface that's going to be shared across all completions not just the chatbot so whos designing that interface because there are two PRs now which have set out to build one obv there cannot be two. 

I feel like it's probably best that @sshivaditya2019 continues to build it maybe and I'll use it? Idk anymore really but lmk if I should build it or wait until #31 has been merged. Thanks.

**Pull request closed without merging**
